File: dependency-inversion-python/src/file_util_inverted.py
from pathlib import Path
from src.path_contract import PathContract

def concatenate_files(source1: PathContract, source2: PathContract, destination: PathContract) -> None:
    bytes1 = source1.read_bytes()
    bytes2 = source2.read_bytes()
    files = bytes1 + bytes2
    destination.write_bytes(files)
    

File: dependency-inversion-typescript/src/fileUtilInverted.ts
import * as fs from "fs";

export function concatenateFiles(
  source1: string, 
  source2: string, 
  destination: string, 
  fsContract: typeof fs
): void {
  const bytes1 = fsContract.readFileSync(source1);
  const bytes2 = fsContract.readFileSync(source2);
  const concatenated = Buffer.concat([bytes1, bytes2]);
  fsContract.writeFileSync(destination, concatenated);
}